src/python/pants/jvm/resolve/jvm_tool.py

```diff
@@ -109,13 +109,12 @@ async def gather_coordinates_for_jvm_lockfile(
     for artifact_input in request.artifact_inputs:
         # Try parsing as a `Coordinate` first since otherwise `AddressInput.parse` will try to see if the
         # group name is a file on disk.
-        if 2 <= artifact_input.count(":") <= 3:
-            try:
-                maybe_coord = Coordinate.from_coord_str(artifact_input).as_requirement()
-                requirements.add(maybe_coord)
-                continue
-            except Exception:
-                pass
+        try:
+            maybe_coord = Coordinate.from_coord_str(artifact_input).as_requirement()
```

Comment:
For your purposes, it looks like you could still require at least 2 colons to trigger this case, which would remove ambiguity.

Reply:
Good point. Will put back the `2 <=` condition then.